src/WebSocketConnection.ts

```diff
@@ -151,6 +151,11 @@ class WebSocketConnection {
   public readonly closedP: Promise<void>;
   protected resolveClosedP: () => void;
 
+  /**
+   * Can reject when if a send fails
+   */
+  protected sendReadyP: Promise<void> | undefined;
+
```

Comment:
If you're going to do this, just use `js-async-locks` instead, we prefer using tooling that we have already engineered to work.

Also how does this impact the benchmark?